packages/slack/src/utils.ts

```diff
@@ -77,68 +86,27 @@ export const welcome = (
 export const channelWelcome =
 	"Hello, I'm Globalping. To learn more about me, run `/globalping help`.";
 
-interface ChannelPayload {
-	channel_id: string;
-	user_id: string;
-	thread_ts?: string;
-}
-
-export const postAPI = async (
-	client: WebClient,
-	payload: ChannelPayload,
-	cmdText: string
-) => {
-	const flags = argsToFlags(cmdText);
-
-	// eslint-disable-next-line @typescript-eslint/naming-convention
-	const { channel_id, user_id, thread_ts } = payload;
-
-	if (!flags.cmd || flags.help) {
-		await client.chat.postEphemeral({
-			text: helpCmd(flags.cmd, flags.target, 'slack'),
-			user: user_id,
-			channel: channel_id,
-			thread_ts,
-		});
-	} else {
-		const postMeasurements = buildPostMeasurements(flags);
-
-		// We post measurement first to catch any validation errors before committing to processing request message
-		logger.debug(`Posting measurement: ${JSON.stringify(postMeasurements)}`);
-		const measurements = await postMeasurement(postMeasurements);
-		await client.chat.postEphemeral({
-			text: '```Processing request...```',
-			user: user_id,
-			channel: channel_id,
-			thread_ts,
-		});
-		logger.debug(`Post response: ${JSON.stringify(measurements)}`);
-		logger.debug(`Latency mode: ${flags.latency}`);
-
-		let first = true;
-		// You can have multiple locations run in parallel
-		for (const measurement of measurements) {
-			const res = await getMeasurement(measurement.id);
-			logger.debug(`Get response: ${JSON.stringify(res)}`);
-			// Only want this to run on first measurement
-			if (first) {
-				await client.chat.postMessage({
-					channel: channel_id,
-					thread_ts,
-					text: `<@${user_id}>, here are the results for \`${cmdText}\``,
-				});
-				first = false;
-			}
+export const getInstallationId = (p: {
+	isEnterpriseInstall: boolean;
+	enterpriseId?: string;
+	teamId?: string;
+}): string => {
+	if (p.isEnterpriseInstall && p.enterpriseId) {
+		return p.enterpriseId;
+	}
+	if (!p.teamId) {
+		throw new Error('No teamId provided');
+	}
+	return p.teamId;
+};
 
-			await measurementsChatResponse(
-				logger,
-				client,
-				channel_id,
-				thread_ts,
-				measurement.id,
-				res,
-				flags
-			);
-		}
+// Slack's user id is not globally unique, so we need to use the combination of installation id and user id
+export const getLocalUserId = (p: {
```

Comment:
It seems this got lost somewhere in the original issue but the login status applies to the whole workspace/installation so this is likely unnecessary. The idea is that a single user (admin) authenticates the organization and all uses (typically employees) can use the bot with the same credentials.

Reply:
yeah, I've missed that